### PR TITLE
Operator timer + with support

### PR DESCRIPTION
**Pull request by bowu**
This commit supports profiling operators' execution time and WITH statements.

### CODE REVIEW

File: libquery/src/Expression.cpp [hunk 1]
```diff
@@ -245,6 +249,14 @@ galois::Expression::ToString() {
   return return_str;
 }
 
+bool
+galois::Expression::IsValid() const {
```
Comment (roshandathathri) [inline libquery/src/Expression.cpp:253]: Such small functions are better suited for the header IMO.

File: libquery/src/CypherCompiler.cpp [hunk 2]
```diff
@@ -925,7 +966,9 @@ galois::CypherCompiler::CompileASTNode(CypherASTNode ast) {
     return result;
   } else if (type == CYPHER_AST_RETURN) {
     galois::gDebug("Compiling AST node return");
```
Comment (roshandathathri) [inline libquery/src/CypherCompiler.cpp:968]: Drop print please

File: libquery/src/OpGraphBuilder.cpp [hunk 3]
```diff
@@ -207,16 +207,22 @@ galois::OpGraphBuilder::Build(
       AddMatchOps(bit_label_schema, &ast_parser);
       break;
     case kASTWith:
-      GALOIS_LOG_DEBUG("Building a WITH op node (TODO)");
-      // TODO(bowu) not for August deadline
+      GALOIS_LOG_DEBUG("Building a WITH op node");
```
Comment (roshandathathri) [inline libquery/src/OpGraphBuilder.cpp:210]: Drop print please

**Review by roshandathathri — APPROVED** [review]
LGTM

Please try to add a couple more slightly different test queries if you can.

File: libquery/src/CypherCompiler.cpp (hunk 2 above)
Comment (bowu) [reply libquery/src/CypherCompiler.cpp:968]: Do we remove all such prints? Loc added them and I think they could be useful.

Comment (roshandathathri) [reply libquery/src/CypherCompiler.cpp:968]: They are too detailed - more like a trace than a debug print. When testing for a new feature, they are reasonable but they pile on quick. Such prints bloat our logs in debug builds. I'm in favor of removing such prints.

**Pull request merged**